internal/services/order/order.go

```diff
@@ -1,13 +1,217 @@
 package order
 
-func BuyOrder(BReq BuyRequest) (bRes BuyResponse, err error) {
-	return bRes, err
+import (
+	"Tradeasy/config"
+	"Tradeasy/internal/model"
+	"Tradeasy/internal/services/stock_exchange"
+	"bytes"
+	"encoding/json"
+	"errors"
+	"github.com/google/uuid"
+	"io/ioutil"
+	"net/http"
+	"time"
+)
+
+func BuyOrder(bReq BuyRequest) (bRes stock_exchange.OrderResponse, err error) {
+	var stocks model.StocksFeed
+	var account model.TradingAccount
+
+	if bReq.BookType == "Market" {
+		if err = config.DB.Table("stocks_feed").Where("stock_name=?", bReq.StockName).Last(&stocks).Error; err != nil {
+			return bRes, err
+		}
+		bReq.LimitPrice = stocks.LTP
+	}
+
+	if err = config.DB.Table("trading_account").Where("user_id=?", bReq.UserId).First(&account).Error; err != nil {
+		return bRes, err
+	}
+
+	if account.Balance < int64(bReq.Quantity*bReq.LimitPrice) {
+		return bRes, errors.New("balance is insufficient for the placed order")
+	}
+
+	account.Balance = account.Balance - int64(bReq.Quantity*bReq.LimitPrice)
+	account.UpdatedAt=time.Now()
+	if err = config.DB.Table("trading_account").Where("user_id=?", bReq.UserId).Updates(&account).Error; err != nil {
+		return bRes, err
+	}
+
+	orderId := uuid.New().String()
+	p := model.PendingOrders{
+		UserId:    bReq.UserId,
+		OrderId:   orderId,
+		StockName: bReq.StockName,
+		OrderType: "Buy",
+		BookType:  bReq.BookType,
+		Quantity:  bReq.Quantity,
+		Status:    status[0],
```

Comment:
Should not be fetched with indexing. Have them defined as constants and maintain all statuses at one place.